### PR TITLE
Added support for checking if company can build from a point.  Now also supports shared networks and full cities.

### PR DESCRIPTION
**Pull request by earlyraven**
Still a bit messy.

### CODE REVIEW

**Comment by earlyraven** [comment]
OK, I've cleaned it up.  I think I'll still be making the changes discussed earlier (Have companies keep track of what companies are a part of their railNetwork, and what cities they are connected to.  I can either update this or do a separate pull request when that's done.

File: src/World.java [hunk 1]
```diff
@@ -125,9 +160,87 @@ public Point getHoverLocation() {
...
+     */
     public boolean canBuildFrom(Point point) {
-        // TODO: Give an actual answer, assuming the builder is the company whose turn it is now
-        return true;
+        Company activeCompany = companies[activeCompanyIndex];
+        return getSharedNetwork(activeCompany).contains(point);
+    }
+
+    private void changeActivePlayer() {
+        activeCompanyIndex++;
```
Comment (sekelsta) [inline src/World.java]: activeCompanyIndex = (activeCompanyIndex + 1) % companies.length;

File: src/World.java [hunk 2]
```diff
@@ -125,9 +160,87 @@ public Point getHoverLocation() {
...
+     * @return true if the active company can start building rails from the specified point, false otherwise.
+     */
     public boolean canBuildFrom(Point point) {
-        // TODO: Give an actual answer, assuming the builder is the company whose turn it is now
-        return true;
+        Company activeCompany = companies[activeCompanyIndex];
+        return getSharedNetwork(activeCompany).contains(point);
+    }
+
+    private void changeActivePlayer() {
```
Comment (sekelsta) [inline src/World.java]: Unused

Comment (earlyraven) [reply src/World.java]: OK, removed.

**Review by sekelsta — APPROVED** [review]

**Pull request merged**